internal/web/server.go

```diff
@@ -0,0 +1,179 @@
+package web
+
+import (
+	"context"
+	"embed"
+	"fmt"
+	"html/template"
+	"net"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/styrainc/regal/internal/explorer"
+	"github.com/styrainc/regal/internal/lsp/cache"
+	"github.com/styrainc/regal/internal/lsp/clients"
+)
+
+//go:embed assets/*
+var assets embed.FS
+
+//nolint:gochecknoglobals,lll
+var tpl = template.Must(template.New("main").Parse(`
```

Comment:
Perhaps embed this template too?

Reply:
I've also embedded htmx now, so that we don't have any online dependencies